cmd/edv-rest/startcmd/start.go

```diff
@@ -434,39 +438,56 @@ func createStartCmd(srv server) *cobra.Command { //nolint: funlen,gocyclo,gocogn
 				return err
 			}
 
-			authType, err := cmdutils.GetUserSetVarFromString(cmd, authTypeFlagName, authTypeEnvKey, true)
+			authTypesCSV, err := cmdutils.GetUserSetVarFromString(cmd, authTypeFlagName, authTypeEnvKey, true)
 			if err != nil {
 				return err
 			}
 
-			authType = strings.ToLower(authType)
+			authTypes := strings.Split(authTypesCSV, ",")
 
-			if authType == "" {
-				authType = noneAuthType
-			}
+			var gnapAuthEnabled, zcapAuthEnabled bool
 
-			if authType != noneAuthType && authType != gnapAuthType && authType != zcapAuthType {
-				return fmt.Errorf("%s is not a valid auth type", authType)
-			}
+			noAuthorizationModeSpecified := len(authTypes) == 1 && authTypes[0] == ""
```

Comment:
these checks look weird and unintuitive
you can use `StringArrayP` when parsing flags, i.e.
```
startCmd.Flags().StringArrayP(authTypeFlagName, "", []string{}, authTypeFlagUsage)
```
and then use more descriptive check like `len(authTypes) == 0`

Reply:
While that would work for the command line parameter, the auth type can also be set via the environment variable `EDV_AUTH_TYPE`, and when I retrieve that I get a string. So ultimately I still end up needing to parse a string into a CSV, so I don't think it makes sense to have two different ways of handling the auth type setting. The strings.Split method always returns at least one element, which is the reason for the sort of weird looking check. I tried giving it a descriptive name (`noAuthorizationModeSpecified`) to hopefully make it more clear.

Reply:
@fqutishat @rolsonquadras - @aholovko was suggesting to me to use the `cmdutils.GetUserSetOptionalVarFromArrayString` method, however I discovered an issue with it that makes it not work as expected with CSV command line parameters. The fix will require me to make a PR in edge-core first, so I talked to @aholovko and we agreed this PR can be merged in as is so as to unblock @rolsonquadras, and in my next PR I can address this issue.